src/promptflow-core/promptflow/_utils/tool_utils.py

```diff
@@ -470,6 +470,53 @@ def _get_function_path(function):
     return func, func_path
 
 
+def validate_groups_if_exist_in_tool_spec(tool):
+    """Validate groups if exist in tool spec."""
+    tool_name = tool.get("name", "")
+    groups = tool.get("groups", "")
+    invalid_group_names = ["advanced"]
+    used_group_names = set()
+    used_inputs = set()
+    for group in groups:
+        group_name = group.get("name", "")
+        group_inputs = set(group.get("inputs", []))
+        # Group must have name and inputs
+        if not group_name or not group_inputs:
+            message_format = "Group must have name and inputs, please check the tool '{0}' and rename the group."
+            raise ValidationException(
+                message=message_format.format(tool_name),
+                message_format=message_format,
+                ErrorTarget=ErrorTarget.TOOL)
+
+        # Some group names cannot be used like advanced
+        if group_name.lower() in invalid_group_names:
+            message_format = "The group name '{0}' cannot be used, please check the tool '{1}' and rename the group."
+            raise ValidationException(
+                message=message_format.format(group_name, tool_name),
+                message_format=message_format,
+                ErrorTarget=ErrorTarget.TOOL)
+
+        # Group name should be unique
```

Comment:
I think duplicate group name may hit yaml load error? Why do we not need such logic for tool input name duplication?